File: Coding_Test/python/위장.py
def solution(clothes):
    answer = 1
    
    # {종류 : 갯수} 딕셔너리 생성
    # 입력(clothes)을 조회하면서 의상의 종류를 key로 넣고 초기 개수를 1로 설정
    # 이미 있는 종류일 경우 value값을 +1 해준다
    clothe = {}
    for i in clothes:
        if i[1] in clothe:
            clothe[i[1]] += 1
        else:
            clothe[i[1]] = 1
            
    # 모든 경우의 수를 구하기 위해 각 종류+1(안입는 경우 포함)의 값을 곱해주고
    # 모두 안 입는 경우인 1을 빼준다
    for i2 in clothe:
        answer *= (clothe[i2]+1)
    return answer - 1

File: Coding_Test/C++/위장.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    
    // <의상의종류, 의상의개수> map 생성
    map<string,int> clothes2;
    
    // 종류에 따른 개수 구하기
    for(int i = 0, n = clothes.size(); i < n; i++){
        auto itr = clothes2.find(clothes[i][1]);                                // 의상 종류를 찾고
        if(itr == clothes2.end()) clothes2.insert(make_pair(clothes[i][1],1));  // 없으면 새로 만들고
        else itr->second++;                                                     // 있으면 개수 증가
    }
    
    // 전체 경우의 수 = (의상1 개수 + 안 입는 경우) X (의상2 + 안 입) X ... X (...) - 1(전부 안 입는 경우 제외)
    for(auto itr = clothes2.begin(); itr != clothes2.end(); ++itr)
    {
        answer = answer * ((itr->second)+1);
    }
    return answer-1;
}
